lib/src/Base/MetaModel/LinearLeastSquares.cxx

```diff
@@ -43,15 +43,9 @@ LinearLeastSquares::LinearLeastSquares()
 /* Constructor with parameters */
 LinearLeastSquares::LinearLeastSquares(const Sample & dataIn,
                                        const Function & inputFunction)
-  : PersistentObject(),
-    dataIn_(dataIn),
-    dataOut_(0, inputFunction.getOutputDimension()),
-    inputFunction_(inputFunction),
-    constant_(inputFunction.getOutputDimension()),
-    linear_(inputFunction.getInputDimension(), inputFunction.getOutputDimension())
+  : LinearLeastSquares(dataIn, inputFunction(dataIn))
```

Comment:
The names `LinearLeastSquares` and `QuadraticLeastSquares` are poorly chosen according to me because it refers a method to solve a problem instead of a way to create a surrogate model. I suggest:
`LeastSquaresLinearPolynomial` and `LeastSquaresQuadraticPolynomial`.

This makes a deeper deprecation, perhaps simpler.